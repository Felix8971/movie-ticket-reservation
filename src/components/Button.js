import React from 'react';
import PropTypes from 'prop-types';

class Button extends React.Component {
  render() {
    return (
      <button className="button" onClick={() => { this.props.handleClick() }}>{this.props.children}</button>
    )
  }
}

Button.propTypes = {
  handleClick: PropTypes.func,    
}

export default Button;