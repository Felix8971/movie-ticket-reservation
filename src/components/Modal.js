import React from 'react';
import PropTypes from 'prop-types';
//import styled from 'styled-components';
import Button from './Button';


class Modal extends React.Component {
  render() {

  // Render nothing if the "show" prop is false
    if (!this.props.msg) {
      return null;
    }

    return (
      <div className='bg-modal' >
        <div className='modal-content'>
          <p>{this.props.msg}</p>
          <Button handleClick={this.props.onCloseModal}>
            Close
          </Button>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  onCloseModal: PropTypes.func.isRequired,    
  msg: PropTypes.string,
}

export default Modal;