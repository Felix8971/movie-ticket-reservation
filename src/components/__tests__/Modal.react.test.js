import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Modal from '../Modal';
import Button from '../Button';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('Modal', () => {

    it('should be defined', () => {
      expect(Modal).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <Modal msg={"Success!"} onCloseModal={()=>{}} />
      );
      expect(tree).toMatchSnapshot();
    });

    it('should not render', () => {
      const tree = shallow(
        <Modal msg={""} onCloseModal={()=>{}} />
      );
      expect(tree.find('div')).toHaveLength(0);
    });

    const mockFn = jest.fn();
    it('should contain a button', () => {
      const tree = shallow(
        <Modal msg={"hey!"} onCloseModal={mockFn} />
      );
      expect(tree.find(Button)).toHaveLength(1);
    })

});