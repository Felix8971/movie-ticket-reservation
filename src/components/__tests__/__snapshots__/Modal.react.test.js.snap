// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    msg="Success!"
    onCloseModal={[Function]}
    show={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="modal-content"
      >
        <p>
          Success!
        </p>
        <Button
          handleClick={[Function]}
        >
          Close
        </Button>
      </div>,
      "className": "bg-modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Success!
          </p>,
          <Button
            handleClick={[Function]}
          >
            Close
          </Button>,
        ],
        "className": "modal-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Success!",
          },
          "ref": null,
          "rendered": "Success!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Close",
            "handleClick": [Function],
          },
          "ref": null,
          "rendered": "Close",
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="modal-content"
        >
          <p>
            Success!
          </p>
          <Button
            handleClick={[Function]}
          >
            Close
          </Button>
        </div>,
        "className": "bg-modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Success!
            </p>,
            <Button
              handleClick={[Function]}
            >
              Close
            </Button>,
          ],
          "className": "modal-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Success!",
            },
            "ref": null,
            "rendered": "Success!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "handleClick": [Function],
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
